provider/gce/google/raw.go

```diff
@@ -152,19 +161,54 @@ func (rc *rawConn) UpdateFirewall(projectID, name string, firewall *compute.Fire
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if err := logOperationErrors(operation); err != nil {
+		return err
+	}
 
-	err = rc.waitOperation(projectID, operation, attemptsLong)
+	err = rc.waitOperation(projectID, operation, attemptsLong, logOperationErrors)
 	return errors.Trace(err)
 }
 
+type handleOperationErrors func(operation *compute.Operation) error
+
+func returnNotFoundOperationErrors(operation *compute.Operation) error {
+	hadOtherErrors := false
+	if operation.Error != nil {
+		for _, err := range operation.Error.Errors {
+			if err.Code == "RESOURCE_NOT_FOUND" {
+				return errors.NotFoundf("resource", err.Message)
```

Comment:
This'll hide subsequent errors, won't it? I think my suggestion of stashing the error in the waitError and exposing it with .Cause() is cleaner. It also means that you don't need both returnNotFound and log variants, so you don't need to pass them into waitOperation.

Reply:
I think you need both but I've updated to use waitError.cause better